logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/ClientRepository.kt

```diff
@@ -214,13 +215,16 @@ class ClientDataSource(
         clientId: ClientId,
         publicKey: ByteArray,
         cipherSuite: CipherSuite
-    ): Either<CoreFailure, Unit> =
+    ): Either<CoreFailure, Unit> = if (clientRegistrationStorage.hasRegisteredMLSClient()) {
```

Comment:
does `clientRegistrationStorage.hasRegisteredMLSClient` take into account if the registered client still exists remotely, meaning if my app session expired and during that time the client was removed remotly, will thisfunction still return true?

Reply:
That's another edge case, if the use was logged out without clearing the data, and delete the client from another device and afterwards tries to login on this device we can't detect it.
What we can do:
Currently if the client is registered, we get this error from the server API:
`InvalidRequestError(errorResponse=ErrorResponse(code=400, message=MLS public key for the given signature scheme already exists, label=mls-duplicate-public-key))`
We can instead of checking the local flag always try to register the client against the server, if we get that error then we ignore it.

Wdyt?

Reply:
For Proteus, we already check if the local client still exists remotely or not, I suggest MLS clients follow the same path as Proteus